### PR TITLE
Read openid configuration from DB first before using config.php

### LINKED ISSUE #194: `auto-provision` does not create database account when enabled.
Like the title says. I have a OIDC provider (KeyCloak) and OwnCloud configured but when I login via OIDC the account does not get created in OwnCloud.

```bash
docker-compose exec owncloud occ config:app:get  openidconnect openid-connect
{
  "provider-url": "https://keycloak.example.com/auth/realms/test_pod_v2",
  "client-id": "owncloud",
  "client-secret": "9be9dc47-38a2-4c92-ab32-063501225515",
  "loginButtonName": "My SSO",
  "search-attribute": "preferred_username",
  "auto-provision": {
    "enabled": true
  }
}
```

OwnCloud login page after redirect from KeyCloak just displayed the text `User with dummyuser is not known.`.

dummyuser

`occ` confirms no user is created.

```bash
docker-compose exec owncloud occ user:list
  - admin: admin
```

KeyCloud config for OwnCloud client.

```json
{
  "id": "37fb4a33-9670-4e63-bb30-55dfa1aa8558",
  "clientId": "owncloud",
  "rootUrl": "https://owncloud.example.com",
  "adminUrl": "https://owncloud.example.com",
  "surrogateAuthRequired": false,
  "enabled": true,
  "alwaysDisplayInConsole": false,
  "clientAuthenticatorType": "client-secret",
  "secret": "**********",
  "redirectUris": [
    "https://owncloud.example.com/*"
  ],
  "webOrigins": [
    "https://owncloud.example.com"
  ],
  "notBefore": 0,
  "bearerOnly": false,
  "consentRequired": false,
  "standardFlowEnabled": true,
  "implicitFlowEnabled": false,
  "directAccessGrantsEnabled": true,
  "serviceAccountsEnabled": false,
  "publicClient": false,
  "frontchannelLogout": false,
  "protocol": "openid-connect",
  "attributes": {
    "id.token.as.detached.signature": "false",
    "saml.assertion.signature": "false",
    "saml.force.post.binding": "false",
    "saml.multivalued.roles": "false",
    "saml.encrypt": "false",
    "oauth2.device.authorization.grant.enabled": "false",
    "backchannel.logout.revoke.offline.tokens": "false",
    "saml.server.signature": "false",
    "saml.server.signature.keyinfo.ext": "false",
    "use.refresh.tokens": "true",
    "exclude.session.state.from.auth.response": "false",
    "oidc.ciba.grant.enabled": "false",
    "saml.artifact.binding": "false",
    "backchannel.logout.session.required": "true",
    "client_credentials.use_refresh_token": "false",
    "saml_force_name_id_format": "false",
    "require.pushed.authorization.requests": "false",
    "saml.client.signature": "false",
    "tls.client.certificate.bound.access.tokens": "false",
    "saml.authnstatement": "false",
    "display.on.consent.screen": "false",
    "saml.onetimeuse.condition": "false"
  },
  "authenticationFlowBindingOverrides": {},
  "fullScopeAllowed": true,
  "nodeReRegistrationTimeout": -1,
  "defaultClientScopes": [
    "web-origins",
    "profile",
    "roles",
    "email"
  ],
  "optionalClientScopes": [
    "address",
    "phone",
    "offline_access",
    "microprofile-jwt"
  ]
}
```

No error logs for either app.

### PR DESCRIPTION
**Pull request by stijnbrouwers**
## Description
Auto-provision is not working when setting the configuration in the database. It only works when you change config.php.
This change is in line with the default behavior for saving OpenID settings (https://doc.owncloud.com/server/next/admin_manual/configuration/user/oidc/oidc.html#save-settings-in-the-database).

## Related Issue
- Fixes #194 (https://github.com/owncloud/openidconnect/issues/194)

## Motivation and Context
Settings the auto-provision using the proposed method (in database) was not working without this change

## How Has This Been Tested?
Log in with a user that not exists resulted in an error (user xxx is not known.)
Applying the changes in this pull request created the user in owncloud.

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
- [X] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by mmattel — APPROVED** [review]

**Comment by mmattel** [comment]
My approval was accidentially, pls do not count on it.

**Comment by DeepDiver1975** [comment]
That change is quite reasonable - thanks for bringing this to our attention!

I'd like to avoid the code duplication. So it might be reasonable to inject Client into the AutoProvisioning server and call getOpenIdConfig - refs https://github.com/owncloud/openidconnect/blob/86c11a575e8f6e5d9f63436141cd45ce6cbc6b76/lib/Client.php#L96

**Comment by stijnbrouwers** [comment]
@DeepDiver1975 totally agree the change is applied.

File: lib/Service/AutoProvisioningService.php [hunk 1]
```diff
@@ -57,21 +58,27 @@ class AutoProvisioningService {
...
 		ILogger $logger,
-		IConfig $config
+		IConfig $config,
+		Client $client
 	) {
 		$this->userManager = $userManager;
 		$this->groupManager = $groupManager;
 		$this->avatarManager = $avatarManager;
 		$this->logger = $logger;
 		$this->config = $config;
```
Comment (DeepDiver1975) [inline lib/Service/AutoProvisioningService.php:79]: since the field config is no longer needed please remove it - TXH a lot! :+1:

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by DeepDiver1975** [comment]
And please have a look at the https://github.com/owncloud/openidconnect/pull/197#issuecomment-999593522 - THX a lot

File: lib/Service/AutoProvisioningService.php (hunk 1 above)
Comment (stijnbrouwers) [reply lib/Service/AutoProvisioningService.php]: Done in commit 'remove unused config var'

**Comment by stijnbrouwers** [comment]
@DeepDiver1975 The comment you are refering to is the CLA right? Is there a nightly processing or something?
I agreed to it but it remains "not signed yet"

**Comment by stijnbrouwers** [comment]
@DeepDiver1975 I am developing from a new machine and there was a typo in my git.config email.
This prevents the CLA to proceed and it's not recommended by Github to change the email of previous commits.
Therefore, I suggest we close this PR and I will open a new one (#198 )

**Pull request closed without merging**